bin/rule_parser.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+import ast
+import yaml
+
+LOGIC_REPLACE = {'||': 'or',
+                 '&&': 'and',
+                 '!=': 'is not',
+                 'true': 'True',
+                 'false': 'False'}
+
+KEYWORDS = ['condition', 'emit', 'delay']
+
+def __process_cond(data):
+    for word in data:
+        if word == "condition":
+            condition = data[word][0]
+            condition = ast.parse(condition).body[0]
+
+        elif word == "emit":
+            # At the moment turn them into methods, eventually
+            # Swap it out with passing to the signaller
+            action = "{}()".format(data[word]["signal"])
+            action = ast.parse(action).body[0]
+
+        else:
+            print("Unhandled keyword {}".format(word))
+
+    ifnode = ast.If(condition.value, [action], [])
+    ast_module = ast.Module([ifnode])
+
+    ast.fix_missing_locations(ast_module)
+
+    code = compile(ast_module, '<string>', 'exec')
+    return code
+
+def parse_rules(data):
+    '''
+        Parse YAML rules for policy manager and return ast code.
+    '''
+
+    data = str(yaml.load(data))
```

Comment:
If all you are doing is a replace on the string, why yaml.load at all? Just read the data off the file normally?